File: katas/7-kyu/speed-control/solution.py
from math import floor
def gps(s, x):
    # your code here, good luck!
    if len(x) <= 1:
        return 0
    max_speed = 0;
    for i in range(1, len(x)):
        dis = x[i] - x[i - 1]
        speed = (3600 * dis) / s
        if max_speed < speed:
            max_speed = speed
    return floor(max_speed)

File: katas/7-kyu/speed-control/solution.c
#include <stdio.h>
int gps(int s, const double x[/*x_len*/], int x_len) {
    if(x_len <= 1){
      return 0;
    }
    int max = 0;
    for(int i = 1; i < x_len; i++){
        double dis = x[i] - x[i - 1];
        double speed = (3600 * dis)/s;
        if(speed > max){
            max = speed;
        }
    }
    return (int)max;
}